### DISCUSSION TITLE
About the `modules.xml`  of `resteasy-spring`

### DISCUSSION THREAD

**Discussion by liweinan** [resteasy/resteasy-spring#179] [Q&A] [answered]
@jamezp I found in the `modules.xml` of `resteasy-spring` it doesn't have dependency of the `springframework` module:

https://github.com/resteasy/resteasy-spring/blob/main/resteasy-spring-feature-pack/common/src/main/resources/modules/system/layers/base/org/jboss/resteasy/resteasy-spring/main/module.xml

In addition, it seems in WildFly dist the modules directory doesn't have the `springframework` module. So I guess this means the deployment project needs to include the spring libs by itself for the module to work right?

**Comment by jamezp** [comment, marked as answer]
We do not want the `org.jboss.resteasy.resteasy-spring` module to have a dependency on the `org.springframework` module. That means it would require the module when the user could be packaging Spring in their deployment. The latter seems to be the standard for Spring based applications.

We definitely cannot add the module dependency in WildFly as we do not and will not ship a Spring module at all.

**Reply by liweinan** [reply]
In addition the `springframework` module defined in the project is just for the testing purpose right?

With my understanding it's used by the `inmodule` tests like:

- https://github.com/resteasy/resteasy-spring/blob/main/testsuite/integration-tests-spring/inmodule/src/test/java/org/jboss/resteasy/test/spring/inmodule/JavaConfigTest.java#L53

And it's needed by the `resteasy-spring-web` module:

- https://github.com/resteasy/resteasy-spring/blob/main/resteasy-spring-feature-pack/common/src/main/resources/modules/system/layers/base/org/jboss/resteasy/resteasy-spring-web/main/module.xml#L31

**Comment by liweinan** [comment]
@jamezp Now I understand, thanks for teaching! :D
